packages/playground/metro.config.js

```diff
@@ -52,13 +51,26 @@ function devResolve(packageName, originDir, moduleName) {
 
   // For each potential extension we need to check for the file in either src and root
   for (const extension of extensions) {
-    const potentialSrcModuleName =
-      path.resolve(originDirSrc, moduleName) + extension;
+    // Start with the src folder
+    let potentialSrcModuleName = path.resolve(originDirSrc, moduleName);
+    if (fs.existsSync(potentialSrcModuleName) &&
+        fs.statSync(potentialSrcModuleName).isDirectory()) {
+      potentialSrcModuleName = path.resolve(potentialSrcModuleName, 'index');
```

Comment:
I dont think this is valid.  finding the index file of a package is more complex than looking for an index file.  There are a bunch of fields in package.json that can specify different index files depending on what kind of JS module you want etc. 

We didn't have code adding index before, why do we need it now?

Reply:
Actually I didn't take into account that this function is only called for files within packages within the repo.  Never for the package root.  So this is probably an ok change.

Still weird that it would be required as part of this change.  It might be more correct though.

Reply:
Bundling still failed, because of our "load from the `src` version if possible" logic. We ask if ​`src/Libraries/NewAppScreen` exists, which it does, then return that as the path for the module. Then metro tries to load the index file there, which doesn't exist, because we only override some sub-paths.